lib/arrays.ncl

```diff
@@ -203,5 +203,23 @@ let error_lib = import "./error.ncl" in
           if has_duplicate then
             error_lib.mk_error "duplicate found: %{std.serialize 'Json duplicate}"
           else
-            'Ok
+            'Ok,
+
+  contract = {
+    minItems
+      : Number -> Dyn
+      | doc m%"
+        A contract that checks whether an array has at least a certain number of items.
+      "%
+      = fun n =>
+        std.contract.from_predicate (fun arr => std.typeof arr == 'Array && std.array.length (arr | Array Dyn) >= n),
```

Comment:
```suggestion
        std.contract.from_predicate (fun arr => std.typeof arr == 'Array && std.array.length (arr | Array Dyn) >= n),
```

Although it's not very expensive, the `Array | Dyn` annotation still has a run time cost, although we know that `arr` is actually an array at this point. I'm wondering if statically typing this function is worth it, especially if it's included automatically in many generated contract (and thus is run potentially many times).

Reply:
I moved the contract annotation to the outer function (removing the static typing). At some point I'll do some performance measurements...